source/apiVolontaria/volunteer/tests/tests_admin_Participation.py

```diff
@@ -0,0 +1,134 @@
+        )
+        self.cycle = Cycle.objects.create(
+            name="my cycle",
+        )
+        self.task_type = TaskType.objects.create(
+            name="my tasktype",
+        )
+
+        start_date = timezone.now() - timezone.timedelta(
+            minutes=100,
+        )
+        end_date = start_date + timezone.timedelta(
+            minutes=50,
+        )
+
+        self.cycle_inactive = Cycle.objects.create(
+            name="my cycle",
+            start_date=start_date,
+            end_date=end_date,
+        )
+
+        # Some date INSIDE the cycle range
+        start_date = start_date + timezone.timedelta(
+            minutes=1,
+        )
+        end_date = end_date - timezone.timedelta(
+            minutes=1,
+        )
+
+        self.event = Event.objects.create(
+            cell=self.cell,
+            cycle=self.cycle,
+            task_type=self.task_type,
+            start_date=start_date,
+            end_date=end_date,
+        )
+
+        self.participation = Participation(event=self.event, user=self.user)
+
+    def create_user(self):
+        self.username = "test_admin"
+        self.password = User.objects.make_random_password()
+        user, created = User.objects.get_or_create(username=self.username)
+        user.set_password(self.password)
+        user.is_staff = True
+        user.is_superuser = True
+        user.is_active = True
+        user.save()
+        self.user = user
+
+        profile = Profile(
+            user=user,
+            phone='1234567890',
+            mobile='0987654321'
+        )
+
+        profile.save()
+
+    def test_admin_access(self):
+        client = Client()
+        client.login(username=self.username, password=self.password)
+
+        admin_pages = [
+            "/admin/volunteer/",
+            "/admin/volunteer/participation/",
+            ]
+
+        for page in admin_pages:
+            resp = client.get(page)
+            self.assertEqual(resp.status_code, 200)
+            self.assertIn("<!DOCTYPE html", str(resp.content))
+
+    def test_admin_participations(self):
+        self.site = AdminSite()
+
+        admin = ParticipationAdmin(Participation, self.site)
+        self.assertNotEqual(admin, None)
+
+        self.assertEqual(admin.user__email(self.participation),
+                         self.participation.user.email)
+        self.assertEqual(admin.user__first_name(self.participation),
+                         self.participation.user.first_name)
+        self.assertEqual(admin.user__last_name(self.participation),
+                         self.participation.user.last_name)
+
+        profile = Profile.objects.filter(
+            user__pk=self.participation.user.pk
+        ).first()
+
+        self.assertEqual(
+            admin.user__phone(self.participation),
+            profile.phone
+        )
+        self.assertEqual(
+            admin.user__mobile(self.participation),
+            profile.mobile
+        )
```

Comment:
Thanks a lot for your contribution! It's almost perfect.

Can you just create a test function for **EACH** method of the admin class in place of a serie of assertEqual in just one function ? This way it's more easy to refactor only one method in the future without refactor a super generic test.

After that it seems ok for me :)